unity-renderer/Assets/UnityGLTF/Scripts/DownloadQueueHandler.cs

```diff
@@ -0,0 +1,117 @@
+using System;
+using UnityEngine;
+
+namespace UnityGLTF
+{
+    internal class DownloadQueueHandler : IDisposable
+    {
+        private static event Action OnCompetitionReset;
+        private static GLTFComponent nearestComponent;
+        private static float nearestDistance = float.MaxValue;
+        private static int competingCount;
+        private static bool isCompetitionFinished = false;
+
+        private bool hasCompeted = false;
+
+        private readonly GLTFComponent gltfComponent;
+        private readonly Camera mainCamera;
+
+        public DownloadQueueHandler(GLTFComponent component)
+        {
+            gltfComponent = component;
+            mainCamera = Camera.main;
+            OnCompetitionReset += OnResetCompetition;
+        }
+
+        public void Dispose()
+        {
+            OnCompetitionReset -= OnResetCompetition;
+
+            if (gltfComponent == null)
+                return;
+
+            if (gltfComponent == nearestComponent)
+            {
+                ResetCompetition();
+            }
+            else if (gltfComponent.isInQueue)
+            {
+                CheckCompetitionFinished();
+            }
+        }
+
+        public bool IsNextInQueue()
+        {
+            //NOTE: start all downloads while renderer is disabled
+            if (mainCamera == null)
```

Comment:
I think these "renderer state" checks would be better handled with the `RendererState` CommonScriptableObject, as we do in other places. We could have both checks in that conditional (I'd check the bool first).

https://github.com/decentraland/unity-renderer/blob/102fb7113e4b7acf8b8ade68112e81c25daf6edc/unity-renderer/Assets/Scripts/MainScripts/DCL/ScriptableObject/CommonScriptableObjects.cs#L153

Reply:
A mild issue with this is that this falls into the GLTF assembly, and we should try to not reference our assemblies from there (I know we do it anyways for some cases, but we should try to avoid it as much as possible).

Reply:
If we don't want to reference our assemblies in the UnityGLTF library then I guess it's fine, ignore my original comment!